src/components/modal/modal_header_title.tsx

```diff
@@ -20,9 +20,14 @@ export const EuiModalHeaderTitle: EuiModalHeaderTitleProps = ({
   ...rest
 }) => {
   const classes = classnames('euiModalHeader__title', className);
+  const childrenWithHeading = useMemo(() => {
+    if (children && !children.hasOwnProperty('$$typeof'))
+      return <h1>{children}</h1>;
+    return children;
+  }, [children]);
   return (
     <div className={classes} {...rest}>
-      {children}
+      {childrenWithHeading}
     </div>
   );
```

Comment:
[continuing convo from https://github.com/elastic/eui/pull/5494#discussion_r772724353]

Ahh actually I totally forgot about `React.isValidElement` (https://davidwalsh.name/react-isvalidelement)! I think it's a bit more readable and exactly what we're looking for (distinguishing JSX vs strings):

```suggestion
  return (
    <div className={classes} {...rest}>
      {React.isValidElement(children) ? children : <h1>{children}</h1>}
    </div>
  );
```

LMK what you think!

Reply:
This feels a lot cleaner, and is the method I was hoping for, but not finding. Works great, and is easy to reason about.